backend/app/expenses/service.py

```
@@ -953,141 +953,203 @@ async def get_user_balance_in_group(
-                # Populate image directly from users collection to avoid extra client round-trips
-                "userImageUrl": user_images.get(friend_id),
-                "netBalance": 0,
-                "owesYou": False,
-                "breakdown": [],
-                "lastActivity": datetime.utcnow(),
+        # Execute aggregation - Single query for all friend balances
+        try:
+            results = await self.settlements_collection.aggregate(pipeline).to_list(
+                length=500
+            )
+        except Exception as e:
```

Ruff (0.14.7)

Do not catch blind exception: `Exception`

(BLE001)